### DISCUSSION TITLE
Cokreate's angular 12 update dependency on Knowledg's angular 12 update

### DISCUSSION THREAD

**Discussion by arun1205** [Sunbird-Knowlg/Community#87] [Bugs]
Context
Co-kreate as a building block needs to upgrade to angular 12 

Problem Statement
In order to implement the above agenda, Co-kreate team found out that cokreate building block internally imports modules from Knowldg BB, which internally has errors. These needs to be fixed or updated to angular 12 itself

Solution
The suggested solution would be to update the common modules that have errors to angular 12. The Ivy engine mandates the imports to be in a specific way. We believe if we do the imports in the way Ivy engine wants, these errors can be mitigated

@mmanimegalai

**Comment by Krishnaj20** [comment]
@pallakartheekreddy - Pls respond to this.

**Reply by HarishGangula** [reply]
@arun1205  Can you share the details of the modules list used from KnowlgBB

**Comment by arun1205** [comment]
@HarishGangula  I believe its  '@sunbird/core', '@sunbird/shared' and '@sunbird/telemetry'
![image](https://user-images.githubusercontent.com/92018797/213654212-26e9ace3-2402-4114-a371-c22e35dcdc24.png)

**Reply by HarishGangula** [reply]
@arun1205  Knowlg doesn't have any packages mentioned, Please check [knowlg microsite](https://knowlg.sunbird.org/learn/readme) for details

**Reply by arun1205** [reply]
https://github.com/orgs/Sunbird-Ed/discussions/333

**Reply by Krishnaj20** [reply]
@arun1205 @HarishGangula - Can you call out the next steps as part of this thread please?

@vinukumar-vs

**Reply by HarishGangula** [reply]
@Krishnaj20  There is nothing to do from Knowlg BB, we can mark this discussion as closed

**Reply by Krishnaj20** [reply]
thanks @HarishGangula .

@arun1205 - which building block needs to unblock you? Is there a discussion thread created in that building block forums to address this?

**Reply by Krishnaj20** [reply]
@arun1205 - are the dependencies resolved?

@vinukumar-vs

**Reply by arun1205** [reply]
@Krishnaj20 There is a dependency on Sunbird-ED as specified earlier https://github.com/orgs/Sunbird-Ed/discussions/333 and not on Knowlg BB
